v1.2/provider/provider-equinix-metal/composition/composition.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: conformancetest.metal.equinix.com
+spec:
+  writeConnectionSecretsToNamespace: upbound-system
+  compositeTypeRef:
+    apiVersion: metal.equinix.com/v1alpha1
+    kind: Conformancetest
+  resources:
+    - base:
+        apiVersion: server.metal.equinix.com/v1alpha2
+        kind: Device
+        spec:
+          forProvider:
+            hostname: crossplane-conformance
+            plan: c3.small.x86
+            operatingSystem: ubuntu_20_04
+            billingCycle: hourly
+            hardware_reservation_id: next_available
+            locked: false
+            networkType: hybrid
+            tags:
+            - crossplane
+          providerConfigRef:
+            name: equinix-metal-provider
+          writeConnectionSecretToRef:
+            name: crossplane-conformance
+            namespace: crossplane-system
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: spec.metro
+          toFieldPath: spec.forProvider.metro
+    - base:
+        apiVersion: vlan.metal.equinix.com/v1alpha1
+        kind: VirtualNetwork
+        spec:
+          forProvider:
+            description: conformance provisioned VLAN
+          providerConfigRef:
+            name: equinix-metal-provider
+      patches:
+        - fromFieldPath: spec.metro
+          toFieldPath: spec.forProvider.metro
+    - base:
+        apiVersion: ports.metal.equinix.com/v1alpha1
+        kind: Assignment
+        spec:
+          forProvider:
+            deviceIdRef:
+              name: crossplane-conformance
```

Comment:
The names have been removed from the original example yaml for the sake of the composition. I'll need to provide this deviceIdRef in some other way.

Reply:
deviceIdSelector is available and provides the necessary composition referencer functionality:

https://github.com/crossplane/crossplane/blob/master/design/design-doc-composition.md#resource-reference-selection

Reply:
The selector is now working and the resources are provisioning successfully.